configure.ac

```diff
@@ -1134,16 +1134,120 @@ char** (*fn)(void* const*, BACKTRACE_RETURN_TYPE) = backtrace_symbols;
    )
 fi
 
+AC_ARG_ENABLE([backtrace],
+              [AS_HELP_STRING([--enable-backtrace=<unwind-ptrace>],
+              [enable printing backtraces.])],
+              [],
+              [enable_backtrace=no])
+if test "x$enable_backtrace" = xyes; then
+   case $my_htop_platform in
+      linux)
+         enable_backtrace=unwind-ptrace
+         ;;
+      *)
+         AC_MSG_ERROR([backtrace feature not implemented for the '$my_htop_platform' platform])
+         ;;
+   esac
+fi
 
-AC_ARG_ENABLE(
-   [hwloc],
-   [AS_HELP_STRING(
-      [--enable-hwloc],
-      [enable hwloc support for CPU affinity; disables affinity support; requires libhwloc @<:@default=no@:>@]
-   )],
-   [],
-   [enable_hwloc=no]
-)
+case "$enable_backtrace" in
+   no)
+      ;;
+   unwind-ptrace)
+      HTOP_PKG_CHECK_MODULES(LIBUNWIND_PTRACE, libunwind-ptrace,
+         [
+            AM_CFLAGS="$AM_CFLAGS $LIBUNWIND_PTRACE_CFLAGS"
+            LIBS="$LIBS $LIBUNWIND_PTRACE_LIBS"
+          ], [
+            AC_MSG_ERROR([can not find required library libunwind-ptrace and libunwind-generic])
+         ]
+      )
+      AC_DEFINE([HAVE_LIBUNWIND_PTRACE], [1], [Define if libunwind-ptrace is present])
+      ;;
+   *)
+      AC_MSG_ERROR([bad value '$enable_backtrace' for --enable-backtrace])
+      ;;
+esac
+if test "x$enable_backtrace" != xno; then
+   AC_DEFINE([HAVE_BACKTRACE_SCREEN], [1], [Define if the backtrace feature is enabled])
+fi
+AM_CONDITIONAL([HAVE_BACKTRACE_SCREEN], [test x"$enable_backtrace" != xno])
+
+if test "x$enable_backtrace" = xunwind-ptrace -a "x$my_htop_platform" = xlinux; then
+   AC_MSG_CHECKING([if libunwind has unw_get_elf_filename])
+   AC_LINK_IFELSE(
+      [AC_LANG_PROGRAM(
+         [[/* unw_get_elf_filename might exist as a macro */
+         #include <libunwind-ptrace.h>]],
+         [[static unw_cursor_t cursor;
+         /* First argument must be non-null */
+         unw_get_elf_filename(&cursor, (char*)0, (size_t)0, (unw_word_t*)0);]]
+      )],
+      [AC_DEFINE([HAVE_LIBUNWIND_ELF_FILENAME], [1], [Define if libunwind has unw_get_elf_filename])
+      AC_MSG_RESULT(yes)],
+      [AC_MSG_RESULT(no)]
+   )
+fi
+
+AC_ARG_ENABLE([demangling],
+              [AS_HELP_STRING([--enable-demangling],
+                              [enable demangling support for backtraces @<:@default=check@:>@])],
+              [],
+              [enable_demangling=check])
+have_libiberty=no
+case $enable_demangling in
+   check|yes|libiberty)
+      AC_CHECK_LIB([iberty], [cplus_demangle], [
+         have_libiberty=yes
+         AC_DEFINE([HAVE_LIBIBERTY], [1], [Define to 1 if libiberty is found.])
+         LIBS="$LIBS -liberty"
+         ], [
+         if test "x$enable_demangling" = xlibiberty; then
+            AC_MSG_ERROR([--enable-demangling specified but libiberty not found])
+         fi
+      ])
```

Comment:
```suggestion
      AC_CHECK_LIB(
         [iberty],
         [cplus_demangle],
         [
            have_libiberty=yes
            AC_DEFINE([HAVE_LIBIBERTY], [1], [Define to 1 if libiberty is found.])
            LIBS="$LIBS -liberty"
         ], [
            if test "x$enable_demangling" = xlibiberty; then
               AC_MSG_ERROR([--enable-demangling specified but libiberty not found])
            fi
         ]
      )
```

Reply:
A small reminder: The order of the libraries (the `-l` options) specified in `LIBS` is significant. If library foo would call functions or otherwise depend on symbols of library bar, then `-lbar -lfoo` will give you a linking error. It must be `-lfoo -lbar` instead. The big rule is that higher level, application libraries should be specified before system libraries. The very last of the list would be `-lc -lgcc`, if the compiler were told not to imply these two libraries.

`AC_CHECK_LIB` would _prepend_ newly checked libraries to `LIBS` for its default action, so the libraries checked in _later_ `AC_CHECK_LIB` calls will appear _earlier_ in the `LIBS` list. This follows the usual order of `configure` that check system libraries first, then check application level libraries.